heartbeat.cgi

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/perl -T
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use lib qw(. lib local/lib/perl5);
+
+use Bugzilla;
+use Bugzilla::Constants;
+use Bugzilla::Error;
+use Bugzilla::Update;
+
+my $ok = eval {
+    my $memcached = Bugzilla->memcached;
+    my $dbh       = Bugzilla->dbh;
+    my $timestamp = $dbh->selectrow_array('SELECT LOCALTIMESTAMP(0)');
+    $memcached->set_config({key => "heartbeat", data => $timestamp});
+    my $mc_timestamp = $memcached->get_config({key => "heartbeat"});
+
+    $mc_timestamp eq $timestamp;
```

Comment:
If two load balancers both call heartbeat.cgi at the same time, spanning a one-second boundary per LOCALTIMESTAMP(0), then they will trigger a race condition and 'NOT OK' below, unless heartbeat.cgi is somehow under a global lock or whatever.

Perhaps ```le``` rather than ```eq``` here would help?